.github/workflows/build-and-push.yaml

```diff
@@ -17,43 +22,46 @@ jobs:
     runs-on: ubuntu-latest
     permissions:
       contents: read
+      deployments: write
       id-token: write
     steps:
       - uses: actions/checkout@v4
       - uses: docker/setup-buildx-action@v2
-      - id: gcp-auth
-        name: Google authentication
-        uses: google-github-actions/auth@v2
-        with:
-          token_format: "access_token"
-          service_account: artifact-writer@${{ env.GCP_PROJECT_ID }}.iam.gserviceaccount.com
-          workload_identity_provider: ${{ vars.GCPV2_GITHUB_WORKLOAD_IDENTITY_PROVIDER }}
-      - name: Log in to the container registry
-        uses: docker/login-action@v2
-        with:
-          registry: ${{ env.GAR_LOCATION }}-docker.pkg.dev
-          username: oauth2accesstoken
-          password: ${{ steps.gcp-auth.outputs.access_token }}
       - name: Run the release script to build version.json file
         working-directory: .
         run: ./script/update_version.sh
-      - name: Build and push app image
-        uses: docker/build-push-action@v3
+      - name: Output version.json
+        run: cat version.json
+      - name: Build Docker image
+        run: make build
+      - name: Build load-test Docker image
+        run: |
+          docker build -t mzcld-demo-load-tests:latest ./load-tests/
+      - name: Run tests in Docker image
+        run: make test
+      - name: Set Docker image tag to sha for updates of the main branch
+        if: github.ref == 'refs/heads/main'
```

Comment:
This will only evaluate to `true` on pushes to `main`, right? Not on PRs to `main`? My assumption is that for this trigger

```
  pull_request:
    branches:
      - main
```

We want to only verify that the container builds and tests pass, not push it (yet)